src/DataModel/IApiDocumentation.ts

```diff
@@ -0,0 +1,9 @@
+import {IMetadataDescription} from "./IMetadataDescription";
+import {IData} from "./IData";
+
+export interface IApiDocumentation extends IMetadataDescription
+{
+    entryPoint: string;
+
+    getEntryPoint(): Promise<IData>;
```

Comment:
Oh so proceduraly. Definitely not what I had in mind ;). But read again and notice that I meant moving the implementation details from the `ApiDocumentation` *class*. My idea was that it shouldn't care about `fetch` and `metadataProvider` etc. It would simply be

``` js
function getEntryPoint() {
  return this.client.getResource(this.entryPoint);
}
```

Otherwise the snippets in you comment above look good to me. Well, maybe I'd make the operation more self-aware, but that's a different discussion

```
var operation = {};
var event;

operation.invoke(event);
```